app/docs/src/reference/components.md

```diff
@@ -0,0 +1,508 @@
+is changed that change will be reflected in Component B.
+
+Let's use the VPC example in AWS again. When creating a `Subnet` component that
+needs a `VpcId` you want that component to subscribe to the `Vpc` component in
+order to obtain the `VpcId` the subnet needs. The `VpcId` value will flow from
+the `Vpc` component into the `Subnet` component whenever it changes.
+
+## Resources
+
+A resource in System Initiative is the data that describes the specific instance
+within the cloud provider. The cloud provider controls the shape of the resource
+data.
+
+Most components can have a resource. When you create a component it will
+initially not have a resource. If you import a component it will have a
+resource. A Component with a resource represents cloud infrastructure in the
+real world. Actions (ADD LINK) update resources within System Initiative and
+apply your changes within SI to your cloud infrastructure.
+
+:::info
+
+Certain Components like "AWS Region" do not have a resource, as it does not map
+to an instance of cloud infrastructure, rather it is a reference for
+pre-existing cloud configuration data.
+
+:::
+
+## Interacting With Components
+
+### Creating a Component
+
+<DocTabs tabs="CLI,AI Agent,Web Application,Public API">
+<TabPanel value="CLI">
+
+To create a Component with the CLI:
+
+```shellscript [Create a Component]
+$ si component create -c 01KC22VWQQSFWN203C83Q42EVZ AWS::EC2::Instance MyNodeEC2Instance
+```
+
+</TabPanel>
+<TabPanel value="AI Agent">
+
+To create a Component with the AI Agent:
+
+```prompt
+> Create a EC2 Instance component named MyNodeEC2Instance
+
+⏺ Perfect! I've successfully created the EC2 Instance component named "MyNodeEC2Instance" in the "Create Component" change set. The component has been configured with:
+
+  - Component ID: 01KC28S0R3W977CNWWNE764M04
+  - Schema: AWS::EC2::Instance
+
+  The component is now ready for you to configure additional properties like:
+  - ImageId (AMI ID)
+  - InstanceType (e.g., t2.micro, t3.small)
+  - KeyName (SSH key)
+  - SecurityGroupIds or SecurityGroups
+  - SubnetId (if using VPC)
+  - UserData (bootstrap scripts)
+```
+
+</TabPanel>
+<TabPanel value="Web Application">
+
+To create a Component:
+
+![Creating a Component](./components/create-component.png)
+
+1. Press the `N` hotkey or click the 'Add a component' button from the bottom
+   bar.
+2. Use the mouse or keyboard to find the Schema you are looking for.
+3. Double click, or hit enter, on the selected Schema.
+
+</TabPanel>
+<TabPanel value="Public API">
+
+:::code-group
+
+```typescript [TypeScript]
+const response = await componentsApi.createComponent({
+  workspaceId,
+  changeSetId,
+  createComponentV1Request: {
+    componentName,
+    schemaName,
+  },
+});
+```
+
+```python [Python]
+TODO
+```
+
+:::
+
+See the [Public API](./public-api) documentation and the SDKs for more details
+on configuring the SDK to communicate with System Initiative.
```

Comment:
I left a similar note on @wendybujalski's PR, but I think we should drop this public API note. The other three methods don't have a similar section. I think it'll be cleaner, more consistent and easier to maintain if the suggested "how do I do this?" code block tabs assume you have your method of choice ready to go.